### DISCUSSION TITLE
Simulation in reduced/generalized coordinate introduces instabilities?

### DISCUSSION THREAD

**Discussion by yzd0014** [google-deepmind/mujoco#1416] [General]
I was reading the documentation of PhysX, it mentions the cons and pros for doing simulations in maximal coordinate and generalized/reduced coordinate. It says that, one problem for reduced coordinate is that it may not be stable. I wonder if somebody knows there are any papers or publications that talk about this in detail since I didn't quite get their explanation. Also I wonder if mujoco did something to tackle the instability issue they mentioned.

<img width="1624" alt="Screenshot 2024-02-12 200935" src="https://github.com/google-deepmind/mujoco/assets/10913570/b758a503-a4e2-4a83-b02a-88253a0f06bd">

**Comment by yuvaltassa** [comment]
Their explanation is not very informative. What I'm guessing they are trying to say is that in reduced coordinates bias forces (Coriolis, centripetal) can be quite complicated and fast-changing, so are hard to integrate accurately. MuJoCo has quite a few solutions to this, most notably the [advanced integrators](https://mujoco.readthedocs.io/en/latest/computation/index.html#numerical-integration) we provide.

PS Velocity clamping is a horrible idea. Never ever do that. Somewhat counter intuitively, it can _inject_ energy into the system and make it go unstable.
